compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt

```diff
@@ -65,6 +77,36 @@ internal class PlatformWindowContext {
         )
     }
 
+    /**
+     * Converts the given [Rect] from the coordinate space of the container window to the coordinate
+     * space of the root UIWindow in which hierarchy with the container window resides.
+     */
+    fun convertContainerWindowRectToRootWindowCGRect(rect: Rect): CValue<CGRect> {
+        val windowContainer = _windowContainer ?: return CGRectZero.readValue()
+
+        val windowContainerWindow = windowContainer.window
+
+        fun UIWindow.density(): Density {
+            return Density(screen.scale.toFloat())
+        }
+
+        return if (windowContainerWindow == null) {
+            if (windowContainer is UIWindow) {
+                // windowContainer is the root window itself
+                rect.toDpRect(windowContainer.density()).asCGRect()
```

Comment:
Why `.systemDensity` does not work here?

Reply:
Because it's wrong and I want to refactor it later.
It always derives the scale from main device screen. iOS supports external displays with possibly different scales, and scale should be retrieved from the window screen.

Reply:
> Because it's wrong and I want to refactor it later.

Let's refactor it later then, but keep the code consistent for now